lib/rspec/core/shell_escape.rb

```diff
@@ -0,0 +1,49 @@
+module RSpec
+  module Core
+    # @private
+    # Deals with the fact that `shellwords` only works on POSIX systems.
+    module ShellEscape
+      module_function
+
+      def quote(argument)
+        "'#{argument.gsub("'", "\\\\'")}'"
+      end
+
+      if RSpec::Support::OS.windows?
+        # :nocov:
+        alias escape quote
+        # :nocov:
+      else
+        require 'shellwords'
+
+        def escape(shell_command)
+          shell_command.shellescape
+        end
+      end
+
+      # Known shells that require quoting: zsh, csh, tcsh.
+      #
+      # Feel free to add other shells to this list that are known to
+      # allow `rspec ./some_spec.rb[1:1]` syntax without quoting the id.
+      #
+      # @private
+      SHELLS_ALLOWING_UNQUOTED_IDS = %w[ bash ksh fish ]
```

Comment:
There's some background to this.  This code originated in #1884.  In that PR, besides simply adding example ids, we also started using them for the first time in the printed "rerun command" that is shown next to failed example to facilitate re-running it for cases where the location isn't unique (such as for a shared example).

Bash, which I believe is far and away the most popular shell (given it's default on OS X and on all linux distros I know of) does not require quoting ids -- you can run `rspec path/to/spec.rb[1:1]` just fine.  It's only some shells -- zsh among them, which I use -- which require quoting the IDs.  We could always quote them but IMO it's a subpar experience to have the extra quoting characters when you're in an environment that doesn't need them, so the consensus in #1884 was to allow the ids to be printed unquoted when the user's shell is one that is known to allow that.